src/3d/qgsoffscreen3dengine.cpp

```diff
@@ -33,31 +33,9 @@
 #include <Qt3DRender/QRenderSurfaceSelector>
 #include <Qt3DRender/QTexture>
 #include <Qt3DRender/QViewport>
-#include <QtGui/QOpenGLContext>
 
 QgsOffscreen3DEngine::QgsOffscreen3DEngine()
 {
-  // Set up the default OpenGL surface format.
-  QSurfaceFormat format;
-
-  // by default we get just some older version of OpenGL from the system,
-  // but for 3D lines we use "primitive restart" functionality supported in OpenGL >= 3.1
-  // Qt3DWindow uses this - requesting OpenGL 4.3 - so let's request the same version.
-#ifdef QT_OPENGL_ES_2
-  format.setRenderableType( QSurfaceFormat::OpenGLES );
-#else
-  if ( QOpenGLContext::openGLModuleType() == QOpenGLContext::LibGL )
-  {
-    format.setVersion( 4, 3 );
-    format.setProfile( QSurfaceFormat::CoreProfile );
-  }
-#endif
-
-  format.setMajorVersion( 3 );
-  format.setDepthBufferSize( 32 ); // TODO: or 24?  (used by QWindow3D)
-  format.setSamples( 8 );
-  QSurfaceFormat::setDefaultFormat( format );
-
```

Comment:
Hmm, the offscreen 3D engine is used in a bunch of other places than just QGIS application (which sets the surface format in main.cpp): in 3D rendering tests and in 3D sandbox executable - all these would need to set the surface format separately. We should probably have a function in qgis_3d that would set up a compatible default format and could be called from various other executables.

Reply:
I did not think about this, yes a dedicated function should work. Otherwise, we could also probably check if the default format has already been set, and set it in qgsoffscreen3dengine if this is not the case.